### DISCUSSION TITLE
contributions are not listed

### DISCUSSION THREAD

**Discussion by kagahd** [prowler-cloud/prowler#1750] [General]
It seems that I'm no longer listed as a contributor, although I've made some contributions that have been merged into main.
Is it on purpose or just an error? Here is the list of my contributions: 
https://github.com/prowler-cloud/prowler/search?q=author%3Akagahd&type=issues

**Comment by toniblyx** [comment]
Hi @kagahd, sorry to hear that, we don't see you in contributors list either. Is it a Github bug? Can you please open a ticket with Github support to let them know? We haven't changed anything in our side, it might be an issue on Github. 

Remember you are still listed in our releases page on every one you have helped out https://github.com/prowler-cloud/prowler/releases

**Reply by kagahd** [reply]
Thanks for your reply @toniblyx. I opened a github support ticket. Let's see what they answer.

**Comment by kagahd** [comment]
Hi @toniblyx, 

Github's support replied the following:

I just had a look at your contributions to the master (default) branch of the prowler-cloud/prowler repository. It seems that you haven't contributed to that branch:

https://github.com/prowler-cloud/prowler/commits?author=kagahd

If you originally were working on a feature branch, they could have been squash merged into the project. Otherwise, they have been rebased or squashed out of existence.

As per [our documentation](https://docs.github.com/en/repositories/viewing-activity-and-data-for-your-repository/viewing-a-projects-contributors#troubleshooting-contributors)

> If you don't appear in a repository's contributors graph, it may be because:
>
>  * You aren't one of the top 100 contributors.
>  * Your commits haven't been merged into the default branch.
>  * The email address you used to author the commits isn't connected to your account on GitHub.
Sorry that's not the answer you were hoping for!

Feel free to come back to me if there's anything else you need.

Best wishes,
Pedro

I always opened prowler pull requests from the feature branch of my forked prowler repo. It seems that these merges into master have been rebased or squashed out of existence since the release of prowler version 3. I well remember being on the prowler contributor list before the release of prowler version 3.
Should I have opened prowler pull requests from the master branch of my forked prowler repo to avoid the issue?
Thanks for your help!

**Comment by kagahd** [comment]
Hi @toniblyx, I just wanted to inform you, that github's support solved the mystery. 

In the previous comment I wrote:
> I always opened prowler pull requests from the feature branch of my forked prowler repo.

That's not true. Most of my pull requests came from the master branch of my forked repo. Therefore, that's not the reason that these commits got vanished from the contributions list.
The reason is, that my commits got **force pushed** or **squashed** into the former `master` branch. At that time, these commits were still counted as my contributions. However, when the former `master` branch became `prowler-2` and the `prowler-3` became the new `master` branch, then my commits got vanished from the contributions history.

According to github's support, you could have avoid this by the use of [Releases](https://docs.github.com/en/enterprise-cloud@latest/repositories/releasing-projects-on-github/about-releases) which are based on git tags, marking different points in your repository's history. Github typically suggest using the [Github flow](https://docs.github.com/en/enterprise-cloud@latest/get-started/quickstart/github-flow), in which changes are made in branches, and ultimately merged into the default branch. This maintains the repository's history, while the default branch is always considered "production ready".

Maybe this can be done in the future to keep the contributions in the history.
Thank you for your attention.

**Reply by toniblyx** [reply]
Hi @kagahd, I'm sorry to hear that changing the master source branch vanish the stats, and that's so weird too, isn't it? In any case I haven't seen much difference in the contributors list after the change we did between versions so the Github support response is kinda weird to me too. In any case, let me know if there is something we can do for you.
